### ISSUE TITLE
User's login session should persist

### ISSUE THREAD

**Issue by cesine** [jrwdunham/dative#82]
There is some stuff we should do to make the users login similar to the backbone app, where if you re-open the app you are back on the same page you were on when you closed it. I'm sure you were planning this but there is some login stuff I should help with since we have the mechanisms for this in place. The user should also be able to open a url /corpusidentifier/docidentifier and land on that pice of data..

I'm realizing that there is some inconsistencies in the logins that we really should deal with before you settle on a login in dative, especially since we need users to be able to work offline in dative.

Prototype:
* users corpora are listed in their user, the corpora can be on any server. we put this into user.corpuses (backbone plurality conventions)

Spreadsheet 
* users corpora are populated by looking at their _session roles from a couchdb server (tobin never used the auth login service to login, he didn't know there was a user doc or that the corpora were listed in the user) we started putting this into user.roles because it was a great way to get the list of corpora the user had permission to access, since the corpuses was out of date because we moved the corpora creation to be both client side and server side

Goal:
* the user can log into any auth server, the user can open any corpus which they have access too (they can provide the corpus server url, or they can add the corpus connection in their user) for this i propose a new objet called user.corpora I will work on making this work as it should adn we can switch to using it later.

**Comment by jrwdunham** [comment]
Yeah, in Dative I'm currently using the in the response returned from POSTing to _session on the corpus service: `roles` is stored in applicationSettings (see models/application-settings.coffee) and then views/corpora.coffee processes these to get the logged-in user's list of accessible corpora, see `getCorpusPouchnames` and `getPouchnameFromRole`.

**Comment by jrwdunham** [comment]
"The user should also be able to open a url /corpusidentifier/docidentifier and land on that pice of data.."

I agree. (The old OLD worked that way too.) 

I've already begun using the Backbone History and router stuff to remember the higher-level pages and connect URL hashes to particular views. I was hoping to use Backbone.history.start({pushState: true}) to get nice hash-less URLs but that seems to throw a wrench into Grunt's livereload. Any ideas about that?

**Comment by jrwdunham** [comment]
"Goal: the user can log into any auth server, the user can open any corpus which they have access too (they can provide the corpus server url, or they can add the corpus connection in their user) for this i propose a new objet called user.corpora I will work on making this work as it should adn we can switch to using it later. (created as a fielddb issue OpenSourceFieldlinguistics/FieldDB#1834)"

Ok, good. From my POV, I'm just wondering if my Dative server config forms/widgets are going to need a new form for the corpus URL. I guess adding a new one is not such a big deal, but I was hoping that it would be enough to specify the Auth URL and then the Auth system could respond (upon successful login) with info about all corpora that the user has access to.

![dative-server-widget](https://cloud.githubusercontent.com/assets/2504271/5888034/1db42f76-a3a4-11e4-9f81-40cdc157c4fd.png)

**Comment by cesine** [comment]
grunt live reload:yes we have it in the psyho Ling dashboard. I'm will add it for you in a branch later

I was feeling artistic in the plane, so I tried to do some wire diagrams and ideal flow documentation which would make a simple user experience, but still permit the user to have corpora on multiple corpus services and also be able to authenticate the user offline.



last image (placed first since it might be the most informative)

![p_20150124_192217](https://cloud.githubusercontent.com/assets/196199/5888224/56e22cb6-a431-11e4-95a8-4587db0413e6.jpg)

![p_20150124_192137](https://cloud.githubusercontent.com/assets/196199/5888239/d0c7b578-a431-11e4-93d8-694e7869e3fd.jpg)



![p_20150124_192144](https://cloud.githubusercontent.com/assets/196199/5888238/c0bf029e-a431-11e4-98b6-7869146416b9.jpg)


![p_20150124_192144](https://cloud.githubusercontent.com/assets/196199/5888236/b74eadfe-a431-11e4-857a-38c78fcdca41.jpg)


![p_20150124_192154](https://cloud.githubusercontent.com/assets/196199/5888234/a2d2e642-a431-11e4-83ae-8620530bf562.jpg)


![p_20150124_192203](https://cloud.githubusercontent.com/assets/196199/5888232/9b6587ac-a431-11e4-8603-e187531a581d.jpg)


![p_20150124_192210](https://cloud.githubusercontent.com/assets/196199/5888230/87c5f132-a431-11e4-93f3-1b9df7ddd30f.jpg)
